torch/csrc/dynamo/eval_frame.c

```diff
@@ -320,6 +320,53 @@ inline static CacheEntry* get_cache_entry(PyCodeObject* code) {
   return extra;
 }
 
+PyObject* _debug_get_cache_entry_list(PyObject* self, PyObject* args) {
+  PyObject* my_object;
+  if (!PyArg_ParseTuple(args, "O", &my_object)) {
+    return NULL;
+  }
+  PyCodeObject* code = (PyCodeObject*)my_object;
+
+  CacheEntry* current_node = get_cache_entry(code);
+
+  PyObject* outer_list = PyList_New(0);
+  if (!outer_list) {
+    return NULL;  // Return NULL if failed to create list
+  }
+  while (current_node != NULL) {
+    // Creating a new Python list for the check_fn and code of current CacheEntry
+    PyObject* inner_list = PyList_New(0);
+    if (!inner_list) {
+      Py_DECREF(outer_list);  // Clean up if failed to create list
+      return NULL;
+    }
+
+    // Add the check_fn and code to the inner list
+    if (PyList_Append(inner_list, current_node->check_fn) < 0) {
+      Py_DECREF(outer_list);
+      Py_DECREF(inner_list);  // Clean up if failed to append
+      return NULL;
+    }
+    if (PyList_Append(inner_list, (PyObject*)current_node->code) < 0) {
+      Py_DECREF(outer_list);
+      Py_DECREF(inner_list);  // Clean up if failed to append
+      return NULL;
+    }
```

Comment:
Might be able to simplify to:
```
PyObject* inner_list = PyTuple_Pack(2, current_node->check_fn, current_node->code);
```

Reply:
Fixed in [e39af20](https://github.com/pytorch/pytorch/pull/106673/commits/e39af20b0dba0fe1f34d5ba95dfe68be74c64762).